packages/server/src/provisioning/oauth.ts

```diff
@@ -0,0 +1,145 @@
+import {
+    AccountProvisioning,
+    BasicProvisioner,
+    BasicProvisionerConfig,
+    Provisioning,
+    ProvisioningStatus,
+} from "@padloc/core/src/provisioning";
+import { getIdFromEmail } from "@padloc/core/src/util";
+import { Storage } from "@padloc/core/src/storage";
+import { ConfigParam } from "@padloc/core/src/config";
+import { request } from "../transport/http";
+import { AccountID } from "@padloc/core/src/account";
+import { Auth } from "@padloc/core/src/auth";
+import { OauthConfig, OauthUserInfo } from "../auth/oauth";
+
+export class OauthProvisionerConfig extends BasicProvisionerConfig {
+    @ConfigParam("number")
+    resyncAfter: number = 10;
+}
+
+export class OauthProvisioner extends BasicProvisioner {
+    constructor(
+        readonly storage: Storage,
+        readonly config: OauthProvisionerConfig,
+        public readonly oauthConfig: OauthConfig
+    ) {
+        super(storage, config);
+    }
+
+    async orgDeleted() {
+        return Promise.resolve();
+    }
+
+    async orgOwnerChanged() {
+        return Promise.resolve();
+    }
+
+    async getProvisioning(opts: { email: string; accountId?: AccountID }) {
+        let provisioning = await super.getProvisioning(opts);
+
+        const auth = await this.storage.get(Auth, await getIdFromEmail(opts.email)).catch(() => null);
+
+        if (auth?.metaData?.oauth) {
+            if (!provisioning.account.metaData) {
+                provisioning.account.metaData = {};
+            }
+            provisioning.account.metaData.oauth = auth.metaData.oauth;
+            auth.metaData.oauth = undefined;
+            await this.storage.save(auth);
+        }
+
+        if (
+            !provisioning.account.metaData?.oauth?.lastSync ||
+            provisioning.account.metaData.oauth.lastSync < Date.now() - this.config.resyncAfter * 1000
+        ) {
+            await this._sync(provisioning);
+        }
+
+        provisioning = await super.getProvisioning(opts);
+
+        provisioning.account.skipTos = true;
+
+        return provisioning;
+    }
+
+    async accountDeleted(_params: { email: string; accountId?: string }): Promise<void> {}
+
+    private async _sync({ account }: Provisioning) {
+        if (!this.oauthConfig) {
+            throw "No oauth configuration found!";
+        }
+
+        if (!account.metaData?.oauth?.tokens) {
+            account.status = ProvisioningStatus.Unprovisioned;
+            account.statusLabel = "Access Denied";
```

Comment:
Is there a reason for the strings in this file to not have been translated?

Reply:
Mostly because I don't expect this class to actually be used in production. It's more of a base class for more advanced  oauth-based provisioning.